src/wizard/index.js

```diff
@@ -1,10 +1,30 @@
 import Character from '../character';
 
-class Wizard {
+class Wizard extends Character{
+  #timesAttacked = 0;
+  #protection = 3;
+
   talk() {
+    return `HELLO! MY NAME IS ${this.name}.`.toUpperCase();
   }
 
   attack() {
+    this.#timesAttacked += 1;
+
+    if (this.#protection > 0) {
+      // When protection spell is active, don't take damage and deal double damage back.
+      this.#protection -= 1;
+      return 2;
+    } else {
+      // Take 10 damage in all other cases
+      if (this.#protection === 0) this.health -= 10;
```

Comment:
While this is technically correct, it's good programming practice to use `>`, `<`, `>=`, `<=` in cases like this. In the instance that something was incorrect in another part of your code and `#protection` was decremented lower than zero, this code would fail.
```suggestion
      if (this.#protection <= 0) this.health -= 10;
```